CMakeLists.txt

```
@@ -0,0 +1,237 @@
+    set(sdl2_net_export_name SDL2_net)
+    set(sdl2_net_install_name_infix shared)
+    set(sdl2_target_name SDL2::SDL2)
+else()
+    set(sdl2_net_export_name SDL2_net-static)
+    set(sdl2_net_install_name_infix static)
+    set(sdl2_target_name SDL2::SDL2-static)
+endif()
+
+sdl_find_sdl2(${sdl2_target_name} ${SDL_REQUIRED_VERSION})
+
+add_library(SDL2_net
+    SDLnet.c
+    SDLnetselect.c
+    SDLnetTCP.c
+    SDLnetUDP.c
+)
+add_library(SDL2_net::${sdl2_net_export_name} ALIAS SDL2_net)
+target_include_directories(SDL2_net PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>"
+)
+target_compile_definitions(SDL2_net PRIVATE
+    BUILD_SDL
+    SDL_BUILD_MAJOR_VERSION=${MAJOR_VERSION}
+    SDL_BUILD_MINOR_VERSION=${MINOR_VERSION}
+    SDL_BUILD_MICRO_VERSION=${MICRO_VERSION}
+)
+target_link_libraries(SDL2_net PRIVATE ${sdl2_target_name})
+if(WIN32)
+    if (MSVC)
+        target_compile_options(SDL2_net PRIVATE /W3 /wd4244)
+    endif()
+    target_compile_definitions(SDL2_net PRIVATE _WINSOCK_DEPRECATED_NO_WARNINGS)
+    target_link_libraries(SDL2_net PRIVATE ws2_32 iphlpapi)
+    if(SDL2NET_BUILD_SHARED_LIBS)
+        target_sources(SDL2_net PRIVATE version.rc)
+    endif()
+endif()
+set_target_properties(SDL2_net PROPERTIES
+    DEFINE_SYMBOL DLL_EXPORT
+    PUBLIC_HEADER SDL_net.h
+    EXPORT_NAME ${sdl2_net_export_name}
+    C_VISIBILITY_PRESET "hidden"
+)
+if(NOT ANDROID)
+    set_target_properties(SDL2_net PROPERTIES
+        DEBUG_POSTFIX "${SDL2NET_DEBUG_POSTFIX}"
+    )
+    if(APPLE)
+        # the SOVERSION property corresponds to the compatibility version and VERSION corresponds to the current version
+        # https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html#mach-o-versions
+        set_target_properties(SDL2_net PROPERTIES
+            SOVERSION "${DYLIB_COMPATIBILITY_VERSION}"
+            VERSION "${DYLIB_CURRENT_VERSION}"
+        )
+    else()
+        set_target_properties(SDL2_net PROPERTIES
+            SOVERSION "${LT_MAJOR}"
+            VERSION "${LT_VERSION}"
+        )
+    endif()
+endif()
+if(SDL2NET_BUILD_SHARED_LIBS AND (APPLE OR (UNIX AND NOT ANDROID)))
+    add_custom_command(TARGET SDL2_net POST_BUILD
+        COMMAND "${CMAKE_COMMAND}" -E create_symlink "$<TARGET_SONAME_FILE_NAME:SDL2_net>" "libSDL2_net$<$<CONFIG:Debug>:${SDL2NET_DEBUG_POSTFIX}>$<TARGET_FILE_SUFFIX:SDL2_net>"
+        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}"
+    )
+endif()
+if(SDL2NET_BUILD_SHARED_LIBS)
+    if(WIN32 OR OS2)
+        set_target_properties(SDL2_net PROPERTIES
+            PREFIX ""
+        )
+    endif()
+    if(OS2)
+        # FIXME: OS/2 Makefile has a different LIBNAME
+        set_target_properties(SDL2_net PROPERTIES
+            OUTPUT_NAME "SDL2net"
+        )
+    elseif(UNIX AND NOT ANDROID)
+        set_target_properties(SDL2_net PROPERTIES
+            OUTPUT_NAME "SDL2_net-${LT_RELEASE}"
+        )
+    endif()
+endif()
```

This is in [a pending pr for SDL2](https://github.com/libsdl-org/SDL/pull/5727), and a to-do for SDL_image.
On MSVC or Watcom targeting Windows or OS/2, the dll import library and the static library have the same output file.
The SDL2 pr fixes this by creating `SDL2-static.lib` for those platforms.
```suggestion
else()
    if(MSVC OR (WATCOM AND (WIN32 OR OS2))
        set_target_properties(SDL2_net PROPERTIES
            OUTPUT_NAME "SDL2_net-static"
        )
    endif()
endif()
```